### PR TITLE
Chore: Fix eslint react hook warnings in grafana-ui

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

Part of #27391

uPlot and GraphNG related fixes seemed too risky. I've tried to fixed them but always ended up going in to some infinite loop so I decided to exlcude them instead.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -11,8 +11,9 @@
   "overrides": [
     {
       "files": [
-        "packages/grafana-ui/**/*/!(*.story).{ts,tsx}",
-        "packages/jaeger-ui-components/**/*.{ts,tsx,js}",
+        "packages/grafana-ui/src/components/uPlot/**/*.{ts,tsx}",
+        "packages/grafana-ui/src/components/GraphNG/GraphNG.tsx",
+        "packages/grafana-ui/src/components/BarChart/BarChart.tsx",
```
Comment (dprokop) [inline .eslintrc:16]: GraphNG and BarChart will shortly say good bye to hooks anyways :) https://github.com/grafana/grafana/pull/30941

**Review by dprokop — APPROVED** [review]
@zoltanbedi sorry for the late review. The platform components seem to work ok, I haven't noticed any regressions.

File: packages/grafana-ui/src/components/NodeGraph/NodeGraph.tsx [hunk 2]
```diff
@@ -87,8 +87,8 @@ export function NodeGraph({ getLinks, dataFrames, nodeLimit }: Props) {
   // TODO we should be able to allow multiple dataframes for both edges and nodes, could be issue with node ids which in
   //  that case should be unique or figure a way to link edges and nodes dataframes together.
   const processed = useMemo(() => processNodes(nodesDataFrames[0], edgesDataFrames[0]), [
-    nodesDataFrames[0],
-    edgesDataFrames[0],
+    edgesDataFrames,
+    nodesDataFrames,
```
Comment (aocenas) [inline packages/grafana-ui/src/components/NodeGraph/NodeGraph.tsx:91]: Not super important but seems weird this is required if explicitly only first element is used. If eslint cannot figure this out lets do:
```
firstEdgesDataFrame = edgesDataFrames[0]
useMemo(..., [firstEdgesDataFrame, ...])
```

**Review by aocenas — APPROVED** [review]
Small nitpick otherwise looks fine

**Pull request merged**